web/src/hooks/useClient.ts

```diff
@@ -0,0 +1,12 @@
+import { useEffect, useState } from "react";
+
+export default function useClient() {
+  // Used to render client-side dependant data like the user's name stored in local storage
+  const [isClient, setIsClient] = useState(false);
+
+  useEffect(() => {
+    setIsClient(true);
+  }, []);
+
+  return isClient;
+}
```

Comment:
Maybe I don't understand this correctly - are you just trying to tell if you're running SSR or actually in the browser? If so, see [this](https://github.com/vercel/next.js/issues/5354#issuecomment-520305040) and don't use a hook.